### PR TITLE
Set format version in packtool

### PR DESCRIPTION
**Pull request by wli3**
the second part of https://github.com/dotnet/cli/issues/8805

I am not sure about the following line. 

```
private static readonly Version _formatVersion = new Version(1,0,0);
```

It is hard coded. However, we change the version number when there is a breaking change in the tools package format, say move folder structure around in the future. When that happens, it is very likely to change this GenerateToolsSettingsFile at the same time. The value is close to code, it is easier to know what it is for. And there is only one place to change.

If we flow this version from, say a prop file. That string "1.0.0" will be passed all the way down. And that prop file is far away from the actual change. We are more likely end up forgetting to update it.

If you think we should flow it from a prop file. Please suggest a good place.

### CODE REVIEW

**Review by peterhuene — APPROVED** [review]

**Comment by peterhuene** [comment]
Given this isn't a package version, I wouldn't personally expect it in a props file.  I'd vote for close to code generating the file.

File: src/Tasks/Microsoft.NET.Build.Tasks/GenerateToolsSettingsFile.cs [hunk 1]
```diff
@@ -4,11 +4,14 @@
 using Microsoft.Build.Framework;
 using System.Xml.Linq;
 using System.IO;
+using System;
 
 namespace Microsoft.NET.Build.Tasks
 {
     public class GenerateToolsSettingsFile : TaskBase
     {
+        private static readonly Version _formatVersion = new Version(1,0,0);
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/GenerateToolsSettingsFile.cs]: Can you comment on when this is expected to change?

**Comment by nguerrera** [comment]
I don't have a huge opinion on this, but if the sole purpose is to mark a breaking boundary for older consumers, we could just use a single number: "1" -> "2" -> "3". 

I just wonder when we're expected to change major, minor, patch if there are 3 parts.

Whatever scheme we use, we should be documenting when to change it.

**Review by nguerrera — CHANGES_REQUESTED** [review]
See above.

File: src/Tasks/Microsoft.NET.Build.Tasks/GenerateToolsSettingsFile.cs (hunk 1 above)
Comment (wli3) [reply src/Tasks/Microsoft.NET.Build.Tasks/GenerateToolsSettingsFile.cs]: will do

**Comment by wli3** [comment]
@nguerrera I added comments and changed version to use int

**Review by nguerrera — APPROVED** [review]

**Comment by wli3** [comment]
@dotnet-bot test this please

**Comment by wli3** [comment]
@MattGertz for tell mode

**Comment by wli3** [comment]
@dotnet-bot test Ubuntu14.04 Release

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT_FullFramework Release

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT_FullFramework Debug

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT Release

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT Debug

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT Debug

**Pull request merged**